librz/util/asn1.c

```diff
@@ -417,46 +404,57 @@ RZ_API char *rz_asn1_to_string(RASN1Object *object, ut32 depth, RzStrBuf *sb) {
 			break;
 		}
 		break;
-	case CLASS_APPLICATION:
+	case RZ_ASN1_CLASS_APPLICATION:
 		snprintf(temp_name, sizeof(temp_name), "Application_%u", object->tag);
 		name = temp_name;
 		break;
-	case CLASS_CONTEXT:
+	case RZ_ASN1_CLASS_CONTEXT:
 		snprintf(temp_name, sizeof(temp_name), "Context [%u]", object->tag); // Context
 		name = temp_name;
 		break;
-	case CLASS_PRIVATE:
+	case RZ_ASN1_CLASS_PRIVATE:
 		snprintf(temp_name, sizeof(temp_name), "Private_%u", object->tag);
 		name = temp_name;
 		break;
 	}
 	if (asn1str) {
 		string = asn1str->string;
 	}
-	if (ASN1_STD_FORMAT) {
+	if (structured) {
 		rz_strbuf_appendf(sb, "%4" PFMT64d "  ", object->offset);
 		rz_strbuf_appendf(sb, "%4u:%2d: %s %-20s: %s\n", object->length,
 			depth, object->form ? "cons" : "prim", name, string);
-		rz_asn1_free_string(asn1str);
+		rz_asn1_string_free(asn1str);
 		if (object->list.objects) {
 			for (i = 0; i < object->list.length; i++) {
-				rz_asn1_to_string(object->list.objects[i], depth + 1, sb);
+				rz_asn1_to_string(object->list.objects[i], depth + 1, structured, sb);
 			}
 		}
 	} else {
-		rz_asn1_print_padded(sb, object, depth, name, string);
-		rz_asn1_free_string(asn1str);
+		asn1_print_padded(sb, object, depth, name, string);
+		rz_asn1_string_free(asn1str);
 		if (object->list.objects) {
 			for (i = 0; i < object->list.length; i++) {
-				RASN1Object *obj = object->list.objects[i];
-				rz_asn1_to_string(obj, depth + 1, sb);
+				RzASN1Object *obj = object->list.objects[i];
+				rz_asn1_to_string(obj, depth + 1, structured, sb);
 			}
 		}
 	}
-	return root ? rz_strbuf_drain(sb) : NULL;
 }
 
-RZ_API void rz_asn1_free_object(RASN1Object *object) {
+RZ_API RZ_OWN char *rz_asn1_to_string2(RZ_NULLABLE RzASN1Object *object, ut32 depth, bool structured) {
```

Comment:
Probably better to rename `rz_asn1_to_string()` instead, e.g. `*_to_strbuf()` or something? @ret2libc what do you think?